From 9bd5ded48e31fe21e41f12afc652370f1c36b6cd Mon Sep 17 00:00:00 2001
From: Sean McBride <sean@rogue-research.com>
Date: Thu, 23 Sep 2021 13:36:23 -0400
Subject: [PATCH 1/8] Fixed -Wnewline-eof warning

---
 kiss_fft_log.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/kiss_fft_log.h b/kiss_fft_log.h
index b5b631a..5012474 100644
--- a/kiss_fft_log.h
+++ b/kiss_fft_log.h
@@ -33,4 +33,4 @@
 
 
 
-#endif /* kiss_fft_log_h */
\ No newline at end of file
+#endif /* kiss_fft_log_h */

From 6a9357514a955729cdd1f9145018c0fff2f30547 Mon Sep 17 00:00:00 2001
From: Sean McBride <sean@rogue-research.com>
Date: Thu, 23 Sep 2021 13:37:21 -0400
Subject: [PATCH 2/8] Fixed -Wmissing-variable-declarations warnings, made
 globals static

---
 tools/psdpng.c | 18 +++++++++---------
 1 file changed, 9 insertions(+), 9 deletions(-)

diff --git a/tools/psdpng.c b/tools/psdpng.c
index 4692302..9899be4 100644
--- a/tools/psdpng.c
+++ b/tools/psdpng.c
@@ -16,15 +16,15 @@
 #include "kiss_fft.h"
 #include "kiss_fftr.h"
 
-int nfft=1024;
-FILE * fin=NULL;
-FILE * fout=NULL;
-
-int navg=20;
-int remove_dc=0;
-int nrows=0;
-float * vals=NULL;
-int stereo=0;
+static int nfft=1024;
+static FILE * fin=NULL;
+static FILE * fout=NULL;
+
+static int navg=20;
+static int remove_dc=0;
+static int nrows=0;
+static float * vals=NULL;
+static int stereo=0;
 
 static
 void config(int argc,char** argv)

From 64b51fa9940d548783624a4ac10f7f10788ab8f5 Mon Sep 17 00:00:00 2001
From: Sean McBride <sean@rogue-research.com>
Date: Thu, 23 Sep 2021 13:37:52 -0400
Subject: [PATCH 3/8] Fixed -Wunused-variable warning, move declaration inside
 #if where it's used

---
 tools/kiss_fastfir.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/tools/kiss_fastfir.c b/tools/kiss_fastfir.c
index d4e666c..02769ff 100644
--- a/tools/kiss_fastfir.c
+++ b/tools/kiss_fastfir.c
@@ -41,7 +41,6 @@ size_t kiss_fastfir( kiss_fastfir_cfg cfg, kffsamp_t * inbuf, kffsamp_t * outbuf
 
 
 
-static int verbose=0;
 
 
 struct kiss_fastfir_state{
@@ -244,6 +243,8 @@ size_t kiss_fastfir(
 #include <sys/mman.h>
 #include <assert.h>
 
+static int verbose=0;
+
 static
 void direct_file_filter(
         FILE * fin,

From aa1ce11a67609fdcc466ef6a6adf3bb5027a2df9 Mon Sep 17 00:00:00 2001
From: Sean McBride <sean@rogue-research.com>
Date: Thu, 23 Sep 2021 13:38:25 -0400
Subject: [PATCH 4/8] Fixed various -Wshorten-64-to-32 warnings

---
 tools/kiss_fastfir.c | 7 ++++---
 tools/psdpng.c       | 2 +-
 2 files changed, 5 insertions(+), 4 deletions(-)

diff --git a/tools/kiss_fastfir.c b/tools/kiss_fastfir.c
index 02769ff..a117865 100644
--- a/tools/kiss_fastfir.c
+++ b/tools/kiss_fastfir.c
@@ -67,7 +67,7 @@ kiss_fastfir_cfg kiss_fastfir_alloc(
     size_t i;
     size_t nfft=0;
     float scale;
-    int n_freq_bins;
+    size_t n_freq_bins;
     if (pnfft)
         nfft=*pnfft;
 
@@ -346,7 +346,7 @@ void do_file_filter(
 
     kiss_fastfir_cfg cfg;
     kffsamp_t *inbuf,*outbuf;
-    int nread,nwrite;
+    size_t nread,nwrite;
     size_t idx_inbuf;
 
     fdout = fileno(fout);
@@ -374,7 +374,8 @@ void do_file_filter(
         nwrite = kiss_fastfir(cfg, inbuf, outbuf,nread,&idx_inbuf) * sizeof(kffsamp_t);
         /* kiss_fastfir moved any unused samples to the front of inbuf and updated idx_inbuf */
 
-        if ( write(fdout, outbuf, nwrite) != nwrite ) {
+        ssize_t written = write(fdout, outbuf, nwrite);
+        if ( written < 0 || (size_t)written != nwrite ) {
             perror("short write");
             exit(1);
         }
diff --git a/tools/psdpng.c b/tools/psdpng.c
index 9899be4..2e05728 100644
--- a/tools/psdpng.c
+++ b/tools/psdpng.c
@@ -121,7 +121,7 @@ void transform_signal(void)
     kiss_fft_cpx *fbuf;
     float *mag2buf;
     int i;
-    int n;
+    size_t n;
     int avgctr=0;
 
     int nfreqs=nfft/2+1;

From c23bdf1af9040a963bfaeac6e5f234f3a828a806 Mon Sep 17 00:00:00 2001
From: Sean McBride <sean@rogue-research.com>
Date: Thu, 23 Sep 2021 13:39:10 -0400
Subject: [PATCH 5/8] Fixed -Wabsolute-value warning, use fabs() instead of
 abs()

---
 tools/psdpng.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/tools/psdpng.c b/tools/psdpng.c
index 2e05728..cd6425f 100644
--- a/tools/psdpng.c
+++ b/tools/psdpng.c
@@ -84,8 +84,8 @@ void val2rgb(float x,rgb_t *p)
 {
     const double pi = 3.14159265358979;
     p->g = (int)(255*sin(x*pi));
-    p->r = (int)(255*abs(sin(x*pi*3/2)));
-    p->b = (int)(255*abs(sin(x*pi*5/2)));
+    p->r = (int)(255*fabs(sin(x*pi*3/2)));
+    p->b = (int)(255*fabs(sin(x*pi*5/2)));
     //fprintf(stderr,"%.2f : %d,%d,%d\n",x,(int)p->r,(int)p->g,(int)p->b);
 }
 

From f3055ee795fd663b8f4c3d2ce303b2391c9661a8 Mon Sep 17 00:00:00 2001
From: Sean McBride <sean@rogue-research.com>
Date: Thu, 23 Sep 2021 13:43:56 -0400
Subject: [PATCH 6/8] Fixed -Wimplicit-float-conversion warnings

---
 tools/kiss_fastfir.c | 2 +-
 tools/psdpng.c       | 6 +++---
 2 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/tools/kiss_fastfir.c b/tools/kiss_fastfir.c
index a117865..62d57c4 100644
--- a/tools/kiss_fastfir.c
+++ b/tools/kiss_fastfir.c
@@ -149,7 +149,7 @@ kiss_fastfir_cfg kiss_fastfir_alloc(
     FFTFWD(st->fftcfg,st->tmpbuf,st->fir_freq_resp);
 
     /* TODO: this won't work for fixed point */
-    scale = 1.0 / st->nfft;
+    scale = 1.0f / st->nfft;
 
     for ( i=0; i < st->n_freq_bins; ++i ) {
 #ifdef USE_SIMD
diff --git a/tools/psdpng.c b/tools/psdpng.c
index cd6425f..3436032 100644
--- a/tools/psdpng.c
+++ b/tools/psdpng.c
@@ -103,7 +103,7 @@ void cpx2pixels(rgb_t * res,const float * fbuf,size_t n)
 
     fprintf(stderr,"min ==%f,max=%f\n",minval,maxval);
     valrange = maxval-minval;
-    if (valrange == 0) {
+    if (valrange == 0.0f) {
         fprintf(stderr,"min == max == %f\n",minval);
         exit (1);
     }
@@ -164,9 +164,9 @@ void transform_signal(void)
             avgctr=0;
             ++nrows;
             CHECKNULL( vals = (float*)realloc(vals,sizeof(float)*nrows*nfreqs) );
-            float eps = 1;
+            float eps = 1.0f;
             for (i=0;i<nfreqs;++i)
-                vals[(nrows - 1) * nfreqs + i] = 10 * log10 ( mag2buf[i] / navg + eps );
+                vals[(nrows - 1) * nfreqs + i] = 10.0f * log10f ( mag2buf[i] / navg + eps );
             memset(mag2buf,0,sizeof(mag2buf[0])*nfreqs);
         }
     }

From 852d8c970dcccdd87feda7dd14bed077bd12fa5c Mon Sep 17 00:00:00 2001
From: Sean McBride <sean@rogue-research.com>
Date: Thu, 23 Sep 2021 13:48:32 -0400
Subject: [PATCH 7/8] Fixed -Wimplicit-int-conversion warnings

---
 tools/psdpng.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/tools/psdpng.c b/tools/psdpng.c
index 3436032..0db99a7 100644
--- a/tools/psdpng.c
+++ b/tools/psdpng.c
@@ -83,9 +83,9 @@ static
 void val2rgb(float x,rgb_t *p)
 {
     const double pi = 3.14159265358979;
-    p->g = (int)(255*sin(x*pi));
-    p->r = (int)(255*fabs(sin(x*pi*3/2)));
-    p->b = (int)(255*fabs(sin(x*pi*5/2)));
+    p->g = (png_byte)(255*sin(x*pi));
+    p->r = (png_byte)(255*fabs(sin(x*pi*3/2)));
+    p->b = (png_byte)(255*fabs(sin(x*pi*5/2)));
     //fprintf(stderr,"%.2f : %d,%d,%d\n",x,(int)p->r,(int)p->g,(int)p->b);
 }
 

From 40c515c0914de98fcb647f7b4c71af232dad1b44 Mon Sep 17 00:00:00 2001
From: Sean McBride <sean@rogue-research.com>
Date: Thu, 23 Sep 2021 13:52:08 -0400
Subject: [PATCH 8/8] Fixed -Wsign-compare warning

---
 tools/psdpng.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/tools/psdpng.c b/tools/psdpng.c
index 0db99a7..0bc8d18 100644
--- a/tools/psdpng.c
+++ b/tools/psdpng.c
@@ -135,13 +135,13 @@ void transform_signal(void)
     while (1) {
         if (stereo) {
             n = fread(inbuf,sizeof(short)*2,nfft,fin);
-            if (n != nfft ) 
+            if (n != (size_t)nfft ) 
                 break;
             for (i=0;i<nfft;++i) 
                 tbuf[i] = inbuf[2*i] + inbuf[2*i+1];
         }else{
             n = fread(inbuf,sizeof(short),nfft,fin);
-            if (n != nfft ) 
+            if (n != (size_t)nfft ) 
                 break;
             for (i=0;i<nfft;++i) 
                 tbuf[i] = inbuf[i];
