From 59c8cdfb0852234a506b7aba3ff8fdf0bde410e6 Mon Sep 17 00:00:00 2001
From: razor_laptop <wangjiaxu67@163.com>
Date: Mon, 18 Oct 2021 00:43:05 +0800
Subject: [PATCH] fixed test/Makefile line 137 missing ')'

---
 test/Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/test/Makefile b/test/Makefile
index 9224c89..c05b3fc 100644
--- a/test/Makefile
+++ b/test/Makefile
@@ -134,7 +134,7 @@ $(BENCHFFTW): benchfftw.c pstats.c
 ifeq ($(LIBFFTW_MISSING), 0)
 	$(CC) -o $@ $(CFLAGS) -DDATATYPE$(KISSFFT_DATATYPE) $^ $(FFTWLIB) -L$(ABS_FFTWLIBDIR) -L.. -l$(KISSFFTLIB_SHORTNAME) -lm
 else
-	$(warning WARNING: No FFTW development files found! FFTW not available for comparison!0
+	$(warning WARNING: No FFTW development files found! FFTW not available for comparison!)
 endif
 
 #
